### DISCUSSION TITLE
Renaming 'public' schema in prisma after introducing multischema support

### DISCUSSION THREAD

**Discussion by fjaenke** [prisma/prisma#18086] [Q&A]
Hi there,
  
Currently we are using Prisma in our project and want to introduce multiple schemas using PostgreSQL's multischema support.  
  
My question is: what is a safe way to rename the default "public" schema to a new custom schema, such as "products"?  

Ideally it should not:
- cause data loss
- require copying data from the "public" schema to the "products" schema
- break the migration history, especially if the project was started without multi-schema support.

Can Prisma migrations be used for this purpose?

**Comment by nikolasburk** [comment]
Hey @fjaenke 👋 thanks a lot for raising this, that's a great question!

I just tried this out locally and was able to successfully rename the schema using the following steps:

1. Rename the schema in plain SQL (I used Postico for this):
    ```sql
    ALTER SCHEMA public 
    RENAME TO products;
    ```
1. Update `url` of the `datasource` in `schema.prisma` by appending `?schema=products`, e.g.: `postgresql://nikolasburk:nikolasburk@localhost:5432/schem-test-1?schema=products`

This worked for me and I was able to run more migrations and query the data afterwards.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
